test/common/config/utility_test.cc

```diff
@@ -188,6 +188,18 @@ TEST(UtilityTest, CheckFilesystemSubscriptionBackingPath) {
   Utility::checkFilesystemSubscriptionBackingPath(test_path);
 }
 
+TEST(UtilityTest, NormalizePath) {
+  EXPECT_EQ("/ok/a/b/c", Utility::normalizePath("/ok/a/b/c"));
+  EXPECT_EQ("/ok/a/b/c", Utility::normalizePath("/ok//a//b//c"));
+  EXPECT_EQ("/ok/a/b/c", Utility::normalizePath("/ok///a///b///c"));
+  EXPECT_EQ("/ok/a/b/c", Utility::normalizePath("////ok/a/b/c"));
+
+  // Maintain the trailing slash.
+  EXPECT_EQ("/ok/a/b/c/", Utility::normalizePath("/ok/a/b/c/"));
+  EXPECT_EQ("/ok/a/b/c/", Utility::normalizePath("///ok/a/b/c///"));
+  EXPECT_EQ("/ok/a/b/c/", Utility::normalizePath("///ok////a////b////c///"));
```

Comment:
I don't think normalization should be this aggressive, messing with the slashes in the middle of the path.

I think the beginning of the path is a special case, and "scheme://origin/" should be treated the same as "scheme://origin".  But I don't think that URL normalizers usually mess with anything after the "/". 

The motivation force in the bug the fixes was really the behavior of replacing prefix "/foo" with "/".  WDYT of handling this at the parsing of that command, where a trailing slash is appended to src or dest if they don't have them?  Then the swap can be done fairly cleanly.

Reply:
+1, that's basically my thinking as well.

Reply:
Sorry for overthinking it. What do you think of handling this “/“ specifically: treating that as replacing with empty string?